extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1367,6 +1482,54 @@ export class Project implements ISqlProject {
 		return firstPropertyElement.childNodes[0].nodeValue!;
 	}
 
+	/**
+	 * Sets the value of project property, overrides value if already present in project file.
+	 * @param propertyName Name of property
+	 * @param propertyValue Value of property
+	 */
+	private async setProjectPropertyValue(propertyName: string, propertyValue: string) {
+		const propertyElements = this.projFileXmlDoc!.documentElement.getElementsByTagName(propertyName);
+
+		let propertyElement: Element | undefined;
+		if (propertyElements.length === 0) {
+			propertyElement = this.addProjectPropertyTag(propertyName);
+		} else {
+			propertyElement = propertyElements[0];
+			if (propertyElement.childNodes.length > 0) {
+				propertyElement.removeChild(propertyElement.childNodes[0]);
+			}
+		}
+
+		propertyElement?.appendChild(this.projFileXmlDoc!.createTextNode(propertyValue));
+		await this.serializeToProjFile(this.projFileXmlDoc);
+	}
+
+	/**
+	 * Adds empty project property tag
+	 * @param propertyTag Tag to add
+	 * @returns Added HTMLElement tag
+	 */
+	private addProjectPropertyTag(propertyTag: string): HTMLElement | undefined {
+		let propertyGroup = this.projFileXmlDoc!.getElementsByTagName(constants.PropertyGroup).item(0);
+		if (propertyGroup === null) {
```

Comment:
Are we always expecting there to be a PropertyGroup in the project file? If so, I can remove this.

Reply:
I don't think so. This will be especially true for SDK-style projects. I think it is fair to have a logic that creates one, if there are none.